types/jquery-monthpicker/index.d.ts

```diff
@@ -0,0 +1,65 @@
+// Type definitions for non-npm package MonthPicker 3.0
+// Project: <https://github.com/KidSysco/jquery-ui-month-picker>
+// Definitions by: doberkofler <https://github.com/doberkofler>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.7
+
+/// <reference types="jquery"/>
+/// <reference types="jqueryui"/>
+
+interface JQueryMonthPickerOptions {
+    AltField?: string|JQuery|Element;
+    AltFormat?: string;
+    Animation?: string;
+    Button?: string|JQuery|Element;
+    ButtonIcon?: string;
+    Disabled?: boolean;
+    Duration?: number|string;
+    HideAnim?: 'fadeOut'|'slideUp'|'none';
+    IsRTL?: boolean;
+    MaxMonth?: Date|number|string;
+    MinMonth?: Date|number|string;
+    MonthFormat?: string;
+    Position?: any;
+    SelectedMonth?: Date|number|string;
+    ShowAnim?: string;
+    ShowIcon?: boolean;
+    ShowOn?: string;
+    StartYear?: number|null;
+    UseInputMask?: boolean;
+    ValidationErrorMessage?: string|null;
+    i18n?: {
+        year?: string;
+        prevYear?: string;
+        nextYear?: string;
+        next12Years?: string;
+        prev12Years?: string;
+        nextLabel?: string;
+        prevLabel?: string;
+        buttonText?: string;
+        jumpYears?: string;
+        backTo?: string;
+        months?: string[];
+    };
+    OnBeforeMenuClose?: () => void;
+    OnBeforeMenuOpen?: () => void;
+    OnAfterMenuClose?: () => void;
+    OnAfterMenuOpen?: () => void;
+}
+
+interface JQuery { // eslint-disable-line @typescript-eslint/no-unused-vars
+    MonthPicker(options: JQueryMonthPickerOptions): JQuery;
+
+    MonthPicker(methodName: 'Clear'): void;
```

Comment:
Instead of disabling the lint, fix the error... this can use the union of literals where return types match

Reply:
I considered doing so, but this would be a lot harder to read and not allow to add any comments. I also noticed that other jQuery definitions specify each method independently. Is there any advantage in using a union of literals that I'm not aware of?

Reply:
When the comments are added you can add the lint disable comments and add the comments. without comments it makes no sense to have overloads. We prefer union over overloads.

Reply:
I have already added the comments.